### PR TITLE
Enh motor bundle

### LINKED ISSUE #435: add motorbundle class
sets the top level hints to be the names of the children that are motors.

### PR DESCRIPTION
**Pull request by tacaswell**
closes #435

### CODE REVIEW

**Comment by danielballan** [comment]
Did someone at NSLS-II break the Debian packages? https://travis-ci.org/NSLS-II/ophyd/jobs/271019267#L665

@klauer -- Any idea what would cause that?

**Comment by klauer** [comment]
Looks like someone changed it just today, breaking a bunch of CI:
`main` has been removed: http://epics.nsls2.bnl.gov/debian/dists/
`wheezy`, `staging` directories have been updated: http://epics.nsls2.bnl.gov/debian/dists/wheezy/

Probably would have been wise to use the distribution name URL to start with, but perhaps as a short-term fix you could ask around and see if someone can symlink them?

**Comment by danielballan** [comment]
Leon:

> I don't see Debian version (code name) in error messages. Release file is inside of directory structure - one in jessie, another in wheez etc. Similar is valid for Ubuntu - if you look at other lines that Stu referred - is has trusty in there, but nothing similar in Debian lines. bnl.gov is a simple mirror from our local repository accessible from controls network. Both look good to me. My take - somebody dropped Debian version from the code that is not happy any more.

**Comment by klauer** [comment]
I think my finger-pointing was premature. Note the double spaces here, highlighted with underlines: `deb http://epics.nsls2.bnl.gov/debian/__main contrib`. EPICS_DEB_DIST isn't set here.

`main`, of course, is just a subdirectory in the distro one.

File: doc/source/device-overview.rst [hunk 1]
```diff
@@ -19,7 +22,28 @@ The base :class:`Device` is not particularly useful on it's own, it
...
+Creating a custom device is as simple as
+
+.. code-block:: python
+
+   from ophyd import Device, EpicsMotor
+   from ophyd import Component as Cpt
+
+   class StageXY(Device):
+       x = Cpt(EpicsMotor, ':X')
+       y = Cpt(EpicsMotor, ':X')
```
Comment (klauer) [inline doc/source/device-overview.rst]: `:Y`

File: doc/source/builtin-devices.rst [hunk 2]
```diff
@@ -47,3 +47,27 @@ devices.
...
+This must be sub-classed (like :class:`~ophyd.device.Device`) to be useful.
+
+.. code-block:: python
+
+   from ophyd import MotorBundle, EpicsMotor
+   from ophyd import Component as Cpt
+
+   class StageXY(MotorBundle):
+       x = Cpt(EpicsMotor, ':X')
+       y = Cpt(EpicsMotor, ':X')
```
Comment (klauer) [inline doc/source/builtin-devices.rst]: `:Y`

**Comment by tacaswell** [comment]
Pseudo-positioner should probably just have it's own version of hints.

**Comment by danielballan** [comment]
@tacaswell Would you move 0fdcc5a to a separate PR? We may want to handle this the new way we are handling caproto (source installs).

**Comment by tacaswell** [comment]
I think I need the other docker for the tests to work.

**Comment by tacaswell** [comment]
try rebasing to make the tests happy

**Pull request merged**